test/mock_sdk.dart

```diff
@@ -160,6 +160,10 @@ class Future<T> {
   factory Future.delayed(Duration duration, [T computation()]) => null;
   factory Future.value([value]) => null;
   static Future wait(List<Future> futures) => null;
+  Future<T> catchError(Function onError, {bool test(Object error)});
+  Future<T> whenComplete(action());
+  Future<T> then(Future<T> onValue(T value), { Function onError });
+  // Original is Future<S> then<S>(FutureOr<S> onValue(T value), { Function onError });
```

Comment:
+1 on making the mock as close to what we have in the SDK.  Feel free to add whatever you need (`FutureOr`, etc).  Thanks!